sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlSensor.java

```diff
@@ -99,62 +84,55 @@ private void runChecks(XmlFile xmlFile) {
           ((AbstractXmlCheck) check).setRuleKey(checks.ruleKey(check));
           ((AbstractXmlCheck) check).validate(sourceCode);
         }
-        saveIssue(sourceCode);
+        saveIssue(context, sourceCode);
 
-        saveSyntaxHighlighting(new XMLHighlighting(xmlFile, fileSystem.encoding()).getHighlightingData(), xmlFile.getInputFile());
+        saveSyntaxHighlighting(context, new XMLHighlighting(xmlFile, fileSystem.encoding()).getHighlightingData(), xmlFile.getInputFile());
       }
     } catch (Exception e) {
       throw new IllegalStateException("Could not analyze the file " + xmlFile.getIOFile().getAbsolutePath(), e);
     }
   }
 
-  private void saveSyntaxHighlighting(List<HighlightingData> highlightingDataList, InputFile inputFile) {
-    Highlightable highlightable = perspective(Highlightable.class, inputFile);
-    if (highlightable != null) {
-      HighlightingBuilder highlightingBuilder = highlightable.newHighlighting();
-
-      for (HighlightingData highlightingData : highlightingDataList) {
-        highlightingBuilder.highlight(highlightingData.startOffset(), highlightingData.endOffset(), highlightingData.highlightCode());
-      }
+  private void saveSyntaxHighlighting(SensorContext context, List<HighlightingData> highlightingDataList, InputFile inputFile) {
+    NewHighlighting highlighting = context.newHighlighting().onFile(inputFile);
 
-      highlightingBuilder.done();
+    for (HighlightingData highlightingData : highlightingDataList) {
+      highlighting.highlight(highlightingData.startOffset(), highlightingData.endOffset(), highlightingData.highlightCode());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this use of "highlight"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)